h2o-algos/src/main/java/hex/glm/DispersionTask.java

```diff
@@ -0,0 +1,630 @@
+
+        public void map(Chunk[] chks) {
+            int chkLen = chks[0].len();
+            int jKIndMax = 0, jKL = 0, jKU = 0, djKL = 0, djKU = 0, d2jKL = 0, d2jKU = 0;
+            double wvMax = 0, dwvMax = 0, d2wvMax = 0, logZ = 0, sumWVj = 0, sumDWVj = 0, sumD2WVj = 0, oneOverSumWVj = 0;
+            _logLL = 0;
+            _dLogLL = 0;
+            _d2LogLL = 0;
+            _nobsLL = 0;
+            _nobsDLL = 0;
+            _nobsD2LL = 0;
+            _nobNegSum = 0;
+            double tempLL = 0, tempDLL = 0, tempD2LL = 0;
+            Map<ConstColNames, Integer> constColName2Ind = new HashMap<>();
+            Map<InfoColNames, Integer> infoColName2Ind = new HashMap<>();
+            setConstIndices(constColName2Ind, _weightPresent ? 3 : 2);
+            setInfoIndices(infoColName2Ind, constColName2Ind.size(), _weightPresent);
+            double weight;
+            for (int rInd = 0; rInd < chkLen; rInd++) {
+                double response = chks[0].atd(rInd);
+                weight = _weightPresent ? chks[WEIGHTIND].atd(rInd) : 1;
+                if (response >= 0) { // this part is only valid for response >= 0
+                    if (response > 0) {
+                        // calculate maximum index of series;
+                        jKIndMax = findMaxTermIndex(chks, rInd, constColName2Ind.get(JMaxConst));
+                        // calculate log(z)
+                        logZ = calLogZ(chks, rInd, constColName2Ind.get(zConst));
+                        // calculate maximum of Wj/Vk, dWj/dVk, d2Wj/dVk2 without derivative constants of (alpha-1)/Phi
+                        // and without 1/y or 1/(PI*y)
+                        wvMax = calLogWVMax(chks, rInd, jKIndMax, logZ);
+                        double logjKIndMax = Math.log(jKIndMax);
+                        dwvMax = wvMax + logjKIndMax;
+                        d2wvMax = dwvMax + logjKIndMax;
+
+                        // locate jL/kL, jU/kU for W/V, dW/dV, d2W/dV2;
+                        jKL = estimateLowerBound(jKIndMax, wvMax, logZ, new EvalLogWVEnv());
+                        jKU = estimateUpperBound(jKIndMax, wvMax, logZ, 0, new EvalLogWVEnv());
+                        if (_calAll) {
+                            djKL = estimateLowerBound(jKIndMax, dwvMax, logZ, new EvalLogDWVEnv());
+                            djKU = estimateUpperBound(jKIndMax, dwvMax, logZ, 1, new EvalLogDWVEnv());
+                            d2jKL = estimateLowerBound(jKIndMax, d2wvMax, logZ, new EvalLogD2WVEnv());
+                            d2jKU = estimateUpperBound(jKIndMax, d2wvMax, logZ, 2, new EvalLogD2WVEnv());
+                        }
+
+                        // sum the series W, dW, d2W but not include 1/y or 1/(PI*y)
+                        sumWVj = sumWV(jKL, jKU, wvMax, logZ, new EvalLogWVEnv());
+                        if (sumWVj <= 0.0)
+                            _nobNegSum++;
+                    }
+                    if (response >= 0 && sumWVj > 0) {
```

Comment:
the `response >= 0` is checked in the outer condition on the line `295`
```suggestion
                    if (sumWVj > 0) {
```